packages/components/src/tooltip/test/index.js

```diff
@@ -169,79 +164,71 @@ describe( 'Tooltip', () => {
 				</Tooltip>
 			);
 
-			const button = wrapper.find( 'button' );
-			button.simulate( 'mouseenter', { type: 'mouseenter' } );
-
-			const popoverBeforeTimeout = wrapper.find( 'Popover' );
+			const button = screen.getByRole( 'button' );
+			fireEvent.mouseOver( button );
+			const popoverBeforeTimeout =
+				container.getElementsByClassName( 'components-popover' );
 			expect( popoverBeforeTimeout ).toHaveLength( 0 );
 			expect( originalMouseEnter ).toHaveBeenCalledTimes( 1 );
 
 			// Popover does not yet exist after default delay, because custom delay is passed.
 			setTimeout( () => {
-				const popoverBetweenTimeout = wrapper.find( 'Popover' );
+				const popoverBetweenTimeout =
+					container.getElementsByClassName( 'components-popover' );
 				expect( popoverBetweenTimeout ).toHaveLength( 0 );
 			}, TOOLTIP_DELAY );
-
 			// Popover appears after custom delay.
 			setTimeout( () => {
-				const popoverAfterTimeout = wrapper.find( 'Popover' );
+				const popoverAfterTimeout =
+					container.getElementsByClassName( 'components-popover' );
 				expect( popoverAfterTimeout ).toHaveLength( 1 );
-
 				jest.runOnlyPendingTimers();
 				jest.useRealTimers();
 			}, 2000 );
 		} );
 
 		it( 'should show tooltip when an element is disabled', () => {
```

Comment:
This test is also testing a lot of implementation details — why do we need to find an event catcher, get its bounding client rect, etc... ?

A user of `Tooltip` is not impacted by any of this. The test could be written in a much simpler way:

- render a tooltip around a disabled button
- make sure that the tooltip text is not displayed by default
- try to hover (or focus) the disabled button
- make sure that the tooltip text is displayed

Reply:
According to [this issue](https://github.com/testing-library/user-event/issues/715), it seems that mouseover on a disabled element does not trigger an event.
Maybe that's why it is written this way 🤔